packages/generator-helper/src/types.ts

```diff
@@ -57,9 +57,16 @@ export type ConnectorType =
   | 'sqlite'
   | 'postgresql'
   | 'sqlserver'
-  | 'jdbc:sqlserver'
   | 'cockroachdb'
 
+  // TODO: this should be removed in favor of `'sqlserver'`, as per `getConfig({ ... }).datasources[0]?.provider` from a schema with `provider = "sqlserver"`
+  // 'jdbc:sqlserver' has been removed in https://github.com/prisma/prisma-engines/pull/2830
+  | 'jdbc:sqlserver'
+
+// TODO: this is also a valid provider as returned by `getConfig({ ... }).datasources[0]?.provider` from a schema with `provider = "postgres"`,
+// but adding it would currently break the TypeScript compilation.
+// | 'postgres'
+
```

Comment:
This type definition could be simplified when https://github.com/prisma/prisma/issues/16201 is solved.